include/zephyr/drivers/sensor.h

```diff
@@ -349,6 +349,32 @@ enum sensor_attribute {
 	 */
 	SENSOR_ATTR_PRIV_START = SENSOR_ATTR_COMMON_COUNT,
 
+
+	/**
+	 * Threshold in milli G to identify an active movement
+	 */
+	SENSOR_ATTR_ACTIVE_THRESH,
+
+	/**
+	 * Threshold in milli G to identify an inactive movement
+	 */
+	SENSOR_ATTR_INACTIVE_THRESH,
+
+	/**
+	 * Threshold in milliseconds to identify an inactive movement
+	 */
+	SENSOR_ATTR_INACTIVE_TIME,
+
+	/**
+	 * Amount of samples in the FIFO to trigger interrupt if exceeds
+	 */
+	SENSOR_ATTR_WATERFALL_LEVEL,
```

Comment:
Please use SENSOR_ATTR_BATCH_DURATION. This isn't a generic enough value. There was a very long discussion about this and since this is a generic API we can't expect applications or library code to know how big the FIFO is. Instead they should set the time they would like to batch data for and the driver can take the current ODR and figure out the appropriate watermark interrupt